apps/os/app/components/organization-switcher.tsx

```diff
@@ -16,21 +16,25 @@ import { SidebarMenu, SidebarMenuButton, SidebarMenuItem } from "./ui/sidebar.ts
 export function OrganizationSwitcher() {
   const trpc = useTRPC();
   const navigate = useNavigate();
-  const currentOrganizationId = useOrganizationId();
-  const loaderData = useRouteLoaderData<typeof orgLoader>("routes/org/loader");
+  const loaderData = useRouteLoaderData<typeof orgLoader>("routes/org/layout");
   const organizationsQuery = useQuery(
     trpc.organization.list.queryOptions(void 0, {
-      initialData: loaderData?.organizations ?? [],
+      initialData: () => loaderData?.organizations ?? [],
+      // Organizations are not likely to change frequently, cache for 5 minutes
+      staleTime: 1000 * 60 * 5,
     }),
   );
-  const userQuery = useQuery(trpc.user.me.queryOptions());
 
-  // Only show organizations where user is owner or member
-  const organizations = organizationsQuery.data.filter(
-    (org) => org.role === "owner" || org.role === "member",
-  );
+  const user = useSessionUser();
 
-  const currentOrganization = organizations.find((org) => org.id === currentOrganizationId);
+  // Only show organizations where user is owner or member
+  const organizations =
+    // TODO: SSR shenanigans, I will find a better solution later
+    (
+      organizationsQuery.data.length > 0
+        ? organizationsQuery.data
+        : (loaderData?.organizations ?? [])
```

Comment:
Why are there two different ways of getting orgs? This will  lead to one way being "wrong" and causing bugs